server/worker/worker.go

```diff
@@ -0,0 +1,338 @@
+	for _, dashboard := range dashboards {
+		if dashboard.Title == "test dashboard - empty" {
+			dashboard.Data["overwrite"] = true
+			dashboard.Data["message"] = "PRO - Auto-add analytics panel"
+
+			payload, err := json.Marshal(dashboard.Data)
+			if err != nil {
+				return
+			}
+
+			res, err := api.Post("/api/dashboards/db", payload)
+			if err != nil {
+				level.Info(api.Logger).Log(
+					"status", "error",
+					"message", "updateDashboards - Failed to update dashboard",
+					"error", err,
+				)
+
+				return
+			}
+
+			var responseAsStruct DashboardUpdateResponse
+			err = json.Unmarshal(res, &responseAsStruct)
+			if err != nil {
+				level.Info(api.Logger).Log(
+					"status", "error",
+					"message", "updateDashboards - Failed to parse update response",
+					"error", err,
+				)
+
+				return
+			}
+
+			if responseAsStruct.Status == "success" {
+				level.Info(api.Logger).Log(
+					"status", "success",
+					"message", "Added analytics to "+dashboard.Title,
+					"error", err,
+				)
+			}
+		}
+	}
+}
+
+func (api *Client) getAnalyticsPanelData(panelId int) map[string]interface{} {
+	return map[string]interface{}{
+		"id":    panelId,
+		"title": "Analytics",
+		"type":  "macropower-analytics-panel",
+		"gridPos": map[string]interface{}{
+			"h": 0,
+			"w": 0,
+			"x": 0,
+			"y": 0,
+		},
+		"options": map[string]interface{}{
+			"analyticsOptions": map[string]interface{}{
+				"dashboard":         "$__dashboard",
+				"flatten":           false,
+				"heartbeatAlways":   false,
+				"heartbeatInterval": 60,
+				"postEnd":           false,
+				"postHeartbeat":     false,
+				"postStart":         true,
+				"server":            api.AnalyticsUrl + "/write",
+			},
+		},
+	}
+}
+
+func (api *Client) AddAnalyticsToDashboards() {
+	res, err := api.Get("/api/search")
+	if err != nil {
+		level.Info(api.Logger).Log(
+			"status", "error",
+			"message", "AddAnalyticsToDashboards - Failed to get dashboard data - /api/search",
+			"error", err,
+		)
+
+		return
+	}
+
+	var response []DashboardsResponse
+	err = json.Unmarshal(res, &response)
+	if err != nil {
+		level.Info(api.Logger).Log(
+			"status", "error",
+			"message", "AddAnalyticsToDashboards - Failed to parse JSON response",
+			"error", err,
+		)
+
+		return
+	}
+
+	var dashboardsToUpdate []Dashboard
+	for _, dashboardEntry := range response {
+		hasAnalyticsPanel := false
+		rawDashboardData := api.GetDashboard(dashboardEntry.Uid)
+		dashboardData := getTypedDashboardData(rawDashboardData)
+		panels := getTypedPanelsData(dashboardData)
+
+		largestPanelId := 0
+		for _, panel := range panels {
+			panelMap, ok := panel.(map[string]interface{})
+			if !ok {
+				continue
+			}
+
+			// Go reports id as float, instead of int
+			panelId := panelMap["id"].(float64)
+			panelIdAsInt := int(panelId)
+			if largestPanelId < panelIdAsInt {
+				largestPanelId = panelIdAsInt
+			}
+
+			if isAnalyticsPanel(panelMap) {
+				hasAnalyticsPanel = true
+
+				break
```

Comment:
isn't this a bug related to detecting `largestPanelId` if after adding analytics panel there is some new panel added?